node/tests/SharedTests.ts

```diff
@@ -1510,6 +1509,32 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `strlen test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = uuidv4();
+                const key1Value = uuidv4();
+                const key1ValueLength = key1Value.length;
+                expect(await client.set(key1, key1Value)).toEqual("OK");
+                expect(await client.strlen(key1)).toEqual(key1ValueLength);
+
+                expect(await client.strlen("nonExistKey")).toEqual(0);
+
+                const listName = "myList";
+                const listKey1Value = uuidv4();
+                const listKey2Value = uuidv4();
+
+                expect(
+                    await client.lpush(listName, [listKey1Value, listKey2Value])
+                ).toEqual(2);
+
+                await expect(client.strlen(listName)).rejects.toThrow();
```

Comment:
Please add a comment explaining what you're testing - that we except data types which aren't' string to fail with strlen